File: extension/host/patternutils.py
def subtractIds(i2, i1):
    is2 = i2.split(".")
    is1 = i1.split(".")
    # TODO: support this
    if len(is2) != len(is1):
        return None
    ds = []
    for i in range(len(is1)):
        ds.append(str(int(is2[i])-int(is1[i])))
    return ".".join(ds)

def addIds(i2, i1):
    is2 = i2.split(".")
    is1 = i1.split(".")
    if len(is1) != len(is2):
        return None
    ds = []
    for i in range(len(is1)):
        ds.append(str(int(is2[i])+int(is1[i])))
    return ".".join(ds)

File: extension/hostjs/patternutils.js
function subtractIds(i2, i1) {
    let is2 = i2.split(".");
    let is1 = i1.split(".");
    // TODO: support this
    if (is1.length != is2.length) {
        return null;
    }
    let ds = [];
    for (let i = 0; i < is1.length; i++) {
        ds.push(parseInt(is2[i]) - parseInt(is1[i]));
    }
    return ds.join(".");
}

function addIds(i2, i1) {
    let is2 = i2.split(".");
    let is1 = i1.split(".");
    if (is1.length != is2.length) {
        return null;
    }
    let ds = [];
    for (let i = 0; i < is1.length; i++) {
        ds.push(parseInt(is2[i]) + parseInt(is1[i]));
    }
    return ds.join(".");
}

module.exports = {
    subtractIds: subtractIds,
    addIds: addIds,
};